### PR TITLE
unsupervised contrastive loss

### PR DESCRIPTION
**Pull request by Dante-Basile**
**Reference issue**

#520 

**Type of change**

This contains the loss function and a basic ResNet50 demo demonstrating training and decreasing loss on CIFAR-10.

**What does this implement/fix?**

This demonstrates that the ResNet50 network can effectively optimize the loss of the unsupervised contrastive loss function. Future experiments will examine the accuracy of the trained network. 

**Additional information**

NDD 2021

### CODE REVIEW

**Review by PSSF23 — CHANGES_REQUESTED** [review]
Refer to file structure in #518

**Review by PSSF23 — CHANGES_REQUESTED** [review]
@Dante-Basile You need to `black` format the files.

@jdey4 do you think this is an ideal way to separate supervised & unsupervised? Or should there be separate folders in `benchmarks`?

**Review by PSSF23 — CHANGES_REQUESTED** [review]
- Your notebooks have unnecessary outputs & empty cells. Remove them.
- Remove commented lines of code

**Review by PSSF23 — APPROVED** [review]
